### PR TITLE
implement TextIOBase encoding attribute

### LINKED ISSUE #3896: io.StringIO doesn't have encoding attribute
`_TextIOBase` is the expected owner of the attribute.

Adding a dummy property `encoding` with value `utf-8` will be a good first issue

### PR DESCRIPTION
**Pull request by hy-kiera**
fix #3896 

- python3
```
>>> import io
>>> io.TextIOBase.encoding
<attribute 'encoding' of '_io._TextIOBase' objects>
```
- RustPython
```
>>>>> import io
>>>>> io.TextIOBase.encoding
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'TextIOBase' object has no attribute 'encoding'
```

### CODE REVIEW

File: vm/src/stdlib/io.rs [hunk 1]
```diff
@@ -701,10 +701,16 @@ mod _io {
...
+    #[derive(Debug, PyPayload)]
     struct _TextIOBase;
 
     #[pyimpl(flags(BASETYPE))]
-    impl _TextIOBase {}
+    impl _TextIOBase {
+        #[pyattr]
+        fn encoding(ctx: &Context) -> PyObjectRef {
+            ctx.none()
+        }
```
Comment (Snowapril) [inline vm/src/stdlib/io.rs:712]:
```suggestion
        #[pyproperty]
        fn encoding(&self, vm: &VirtualMachine) -> PyObjectRef {
            vm.ctx.none()
        }
```
For adding `encoding` as getset_descriptor, we need to use `#[pyproperty]`

Comment (hy-kiera) [reply vm/src/stdlib/io.rs:712]: When I recomplied your suggestion, I cannot get `None` value. I'm confusing `pyproperty` and `pyattr`.
- python3
```
>>> print(io.TextIOBase().encoding)
None
>>> print(io.TextIOBase.encoding)
<attribute 'encoding' of '_io._TextIOBase' objects>
```
- rustpython
```
>>>>> print(io.TextIOBase().encoding)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Expected payload '_TextIOBase' but 'TextIOBase' found
>>>>> print(io.TextIOBase.encoding)
<getset_descriptor object at 0x60000341f480>
```

Comment (youknowone) [reply vm/src/stdlib/io.rs:712]: `pyproperty` creates a getset field, which is very similar to `property` in python.
`pyattr` creates a simple attribute, same to `self.encoding = None`

Comment (youknowone) [reply vm/src/stdlib/io.rs:712]: @Snowapril is it a getset instead of attribute?

Comment (Snowapril) [reply vm/src/stdlib/io.rs:712]: Yes. It should be `getset`. `getset_repr` will print "<attribute '%V' of '%s' objects>".
Current rustpython getset does not implement `repr`.

**Review by youknowone — APPROVED** [review]
looks good, thank you!

**Pull request merged**